generated/java/build.gradle

```diff
@@ -99,7 +99,62 @@ task allDocs(type: Javadoc) {
 
   source subProjects.collect {it.sourceSets.main.allJava}
   classpath = files(subProjects.collect {it.sourceSets.main.compileClasspath})
-  destinationDir = file("${buildDir}/all/latest/apidocs")
+  destinationDir = file("${buildDir}/javadoc/all/latest/apidocs")
 
   options.links += ["https://developers.google.com/protocol-buffers/docs/reference/java/"]
 }
+
+task checkOutGhPages {
+  doLast {
+    exec {
+      commandLine "git", "clone", "--branch", "gh-pages", "--single-branch", "git@github.com:googleapis/googleapis.git", "tmp_gh-pages"
+    }
+  }
+}
+
+task copyFilesToGhPages {
+  dependsOn 'allDocs'
+  dependsOn 'checkOutGhPages'
+
+  doLast {
+    def newSiteDirPath = "tmp_gh-pages/java/"
+    new File(newSiteDirPath).mkdirs()
+    copy {
+      from "${buildDir}/javadoc"
+      into newSiteDirPath
+    }
+  }
+}
+
+task commitDocs {
+  dependsOn 'copyFilesToGhPages'
+
+  doLast {
+    def gitHashOut = new ByteArrayOutputStream()
+    exec {
+        commandLine "git", "rev-parse", "HEAD"
+        standardOutput = gitHashOut
+    }
+    def gitHash = gitHashOut.toString().trim()
+
+    exec {
+      workingDir "tmp_gh-pages"
+      commandLine 'git', 'add', '.'
+    }
+
+    exec {
+      workingDir "tmp_gh-pages"
+      commandLine 'git', 'commit', '-m', "regenerate javadocs\n\napi-client-staging: ${gitHash}"
+    }
+  }
+}
+
+task pushDocs {
```

Comment:
Can we add removal of tmp_gh-pages to the clean task?

Reply:
done